### PR TITLE
Refactor `check_logical_slots_readiness` split to reduce complexity

### PR DESCRIPTION
**Pull request by matthbakeredb**
Includes:
* renaming of `_unready_logical_slots` to better represent that it is a processing queue which is emptied on successful completion.
* ensuring that return type is consistent.
* made logic variable names explicit to help explain how the decision of whether a slot is "ready" is made.

Split from #2743

### CODE REVIEW

File: patroni/postgresql/slots.py [hunk 1]
```diff
@@ -331,48 +332,89 @@ def _get_leader_connection_cursor(self, leader: Leader) -> Generator[Union['curs
...
+
+        :param catalog_xmin: ``catalog_xmin`` of the physical slot used by this replica to stream changes from primary.
+        :param slots: dictionary of slot information from the primary
+
+        :returns: ``False`` if any issue was faced while processing, ``True`` otherwise.
+        """
+        if catalog_xmin is not None and cluster.leader:
+            for name, value in slots.items():
+                self.logical_slots_processing_queue[name] = value
+        else:  # Replica isn't streaming or the hot_standby_feedback isn't enabled
```
Comment (CyberDem0n) [inline patroni/postgresql/slots.py:377]: Here we can make the code a bit more linear:
```suggestion
            return True
        # Replica isn't streaming or the hot_standby_feedback isn't enabled
```
Of course, the try...except block should be indented back to the left.

File: patroni/postgresql/slots.py [hunk 2]
```diff
@@ -331,48 +332,89 @@ def _get_leader_connection_cursor(self, leader: Leader) -> Generator[Union['curs
...
+
+        :param primary_physical_catalog_xmin: is the value retrieved from ``pg_catalog.pg_get_replication_slots()`` for
+                                              the physical replication slot on the primary.
+        """
+        # Make a copy of processing queue keys as a list as the queue dictionary is modified inside the loop.
+        for name in list(self.logical_slots_processing_queue):
+            primary_logical_catalog_xmin = self.logical_slots_processing_queue[name]
+            standby_logical_slot = self._replication_slots.get(name, {})
+            standby_logical_catalog_xmin = standby_logical_slot.get('catalog_xmin', 0)
+            if TYPE_CHECKING:
```
Comment (CyberDem0n) [inline patroni/postgresql/slots.py:406]:
```suggestion
            if TYPE_CHECKING:  # pragma: no cover
```

File: patroni/postgresql/slots.py [hunk 3]
```diff
@@ -331,48 +332,90 @@ def _get_leader_connection_cursor(self, leader: Leader) -> Generator[Union['curs
...
+                logger.error("Failed to check %s physical slot on the primary: %r", slot_name, e)
+                return False
+
+            if not self._update_pending_logical_slot_primary(cluster, slots, catalog_xmin):
+                return False  # since `catalog_xmin` isn't valid further checks don't make any sense
+
+        self._ready_logical_slots(catalog_xmin)
+        return True
+
+    def _update_pending_logical_slot_primary(self, cluster: Cluster,
```
[the remark this answers is not in the document]
Comment (matthbakeredb) [reply patroni/postgresql/slots.py]: Ah yes! Thanks. Fixed.

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**